python/lsst/afw/detection/multiband.py

```diff
@@ -0,0 +1,378 @@
+        Images to extract the footprint from
+    thresh: float
+        All pixels above `thresh` will be included in the footprint
+    xy0: `Point2I`
+        Location of the minimum value of the images bounding box
+        (if images is an array, otherwise the image bounding box is used).
+
+    Returns
+    -------
+    spans: SpanSet
+        Union of all spans in the images above the threshold
+    imageBBox: Box2I
+        Bounding box for the input images.
+    """
+    # Set the threshold for each band
+    if not hasattr(thresh, "__len__"):
+        thresh = [thresh] * len(images)
+
+    # If images is a list of `afw Image` objects then
+    # merge the SpanSet in each band into a single Footprint
+    if isinstance(images, MultibandBase) or isinstance(images[0], Image):
+        spans = SpanSet()
+        for n, image in enumerate(images):
+            mask = image.array > thresh[n]
+            mask = Mask(mask.astype(np.int32), xy0=image.getBBox().getMin())
+            spans = spans.union(SpanSet.fromMask(mask))
+        imageBBox = images[0].getBBox()
+    else:
+        # Use thresh to detect the pixels above the threshold in each band
+        thresh = np.array(thresh)
+        if xy0 is None:
+            xy0 = Point2I(0, 0)
+        mask = np.any(images > thresh[:, None, None], axis=0)
+        mask = Mask(mask.astype(np.int32), xy0=xy0)
+        spans = SpanSet.fromMask(mask)
+        imageBBox = mask.getBBox()
+    return spans, imageBBox
+
+
+def heavyFootprintToImage(heavy, fill=np.nan, bbox=None, imageType=MaskedImage):
+    """Create an image of a HeavyFootprint
+
+    Parameters
+    ----------
+    heavy: `HeavyFootprint`
+        The HeavyFootprint to insert into the image
+    fill: number
+        Number to fill the pixels in the image that are not
+        contained in `heavy`.
+    bbox: `Box2I`
+        Bounding box of the output image.
+    imageType: type
+        This should be either a `MaskedImage` or `Image` and describes
+        the type of the output image.
+
+    Returns
+    -------
+    image: `MaskedImage` or `Image`
+        An image defined by `bbox` and padded with `fill` that
+        contains the projected flux in `heavy`.
+    """
+    if bbox is None:
+        bbox = heavy.getBBox()
+    image = imageType(bbox, dtype=heavy.getImageArray().dtype)
+    image.set(fill)
+    heavy.insert(image)
+    return image
+
+
+class MultibandFootprint(MultibandBase):
+    """Multiband Footprint class
+
+    A `MultibandFootprint` is a collection of HeavyFootprints that have
+    the same `SpanSet` and `peakCatalog` but different flux in each band.
+
+    Parameters
+    ----------
+    filters: list
+        List of filter names.
+    footprint: `Footprint`
+        `Footprint` that contains the `SpanSet` and `PeakCatalog`
+        to use for the `HeavyFootprint` in each band.
+    mMaskedImage: `MultibandMaskedImage`
+        MultibandMaskedImage that footprint is a view into.
```

Comment:
Ok, I've updated `__init__` to accept a list of single band `HeavyFootprints` and added a constructor `fromMaskedImages`. It might be worth opening a ticket (or RFC) if necessary to add a constructor to `HeavyFootprint` that allows the user to pass a 1D array (or reference to a 1D array). This is useful for both `MultibandFootprints` and copying footprints, where a shallow copy should still point to the same 1D pixel array, not just the same peak catalog.

Reply:
I definitely agree that that constructor would be nice to have.  It should not require an RFC, as we ought to be able to do it without breaking any existing APIs, and it's not a major addition.

Reply:
Ok, I'll open a ticket.